src/sentry/notifications/helpers.py

```diff
@@ -583,3 +589,29 @@ def get_values_by_provider(
         get_value_for_actor(notification_settings_by_scope, recipient),
         get_value_for_parent(notification_settings_by_scope, parent_id, type),
     )
+
+
+def get_providers_for_recipient(recipient: User | Team) -> Iterable[ExternalProviders]:
```

Comment:
Would you be able to use one of the existing methods like `where_should_recipient_be_notified`? I guess that one takes a single type ... it seems weird that we don't already have this though. Or maybe `transform_to_notification_settings_by_recipient`?

Reply:
@ceorourke No, I cannot use that for the fundamental reason that `where_should_recipient_be_notified` doesn't check the `Identity` table. What might make more sense, in the long run, is for `where_should_recipient_be_notified` to call `get_providers_for_recipient` and use that to determine how we should notify the user.